doc/articles/Asynchronous Operations.md

```diff
@@ -0,0 +1,68 @@
+# Asynchronous Operations {#Asynchronous-Operations}
+
+Asynchronous operations in webgpu.h return a @ref WGPUFuture. This is an opaque handle which applications may use to poll or wait for completion of the asynchronous operation.
+
+On all implementations of webgpu.h, it must be possible to react to async operations without undue latency or spin-waiting.
+
+## Creation
+
+All asynchronous operations start when the application calls an asynchronous webgpu.h function. These functions return `WGPUFuture`, and take in their arguments a `CallbackInfo` struct like @ref WGPURequestDeviceCallbackInfo or @ref WGPUBufferMapCallbackInfo. `CallbackInfo` structs have the following members:
+ - a `WGPUChainedStruct const * nextInChain` pointer, for extensions.
+ - a @ref WGPUCallbackMode `mode` enum.
+ - a `callback` function pointer. For example:
+   ```typedef void (*WGPUBufferMapCallback)(
+        WGPUMapAsyncStatus status,
+        char const * message,
+        WGPU_NULLABLE void* userdata1,
+        WGPU_NULLABLE void* userdata2) WGPU_FUNCTION_ATTRIBUTE;```
```

Comment:
This seems to work in Doxygen but in GitHub it chops off the first line because it parses it as a syntax marker. An actual syntax marker seems to work in both Doxygen and Markdown, and adds syntax highlighting in Doxygen too.

```suggestion
    ```c
    typedef void (*WGPUBufferMapCallback)(
    WGPUMapAsyncStatus status,
    char const * message,
    WGPU_NULLABLE void* userdata1,
    WGPU_NULLABLE void* userdata2) WGPU_FUNCTION_ATTRIBUTE;
    ```
```

Reply:
Done